packages/appsync-modelgen-plugin/src/__tests__/visitors/appsync-swift-visitor.test.ts

```diff
@@ -368,6 +382,60 @@ describe('AppSyncSwiftVisitor', () => {
     `);
   });
 
+  it('Should produce same result for @primaryKey as when @key is used for a primary key', () => {
+    const schemaV1 = /* GraphQL */ `
+      type authorBook @model @key(fields: ["author_id"]) {
+        id: ID!
+        author_id: ID!
+        book_id: ID!
+        author: String
+        book: String
+      }
+    `;
+    const schemaV2 = /* GraphQL */ `
+      type authorBook @model {
+        id: ID!
+        author_id: ID! @primaryKey
+        book_id: ID!
+        author: String
+        book: String
+      }
+    `;
+    const visitorV1 = getVisitor(schemaV1, 'authorBook');
+    const visitorV2 = getVisitorPipelinedTransformer(schemaV2, 'authorBook');
+    const version1Code = visitorV1.generate();
+    const version2Code = visitorV2.generate();
+
+    expect(version1Code).toMatch(version2Code);
```

Comment:
The test should also compare the `CodeGenGenerateEnum.metadata` type in addition to code type

Reply:
I suppose the new test will fail due to the first CR

Reply:
Added